loader:
  pcds_dir: "/tmp/dump/frames"
  pose_file: "/tmp/dump/traj_lidar.txt"
  kitti_calibration_file: ""
  start: 0
  end: 1000
  format: "glim" #"kitti" #"erasor"

map_cleaner:
  save_dir: "/tmp"
  frame_id: "map"
  in_process_visualization: true
  visualization_vg_res: 0.4

ground_segmentation:
  use_voxel_grid: true
  voxel_leaf_size: 0.1
  frame_skip: 0

patchworkpp:
  verbose: false
  enable_rnr: true
  enable_rvpf: true
  enable_tgr: true
  num_iter: 3
  num_lpr: 20
  num_min_pts: 10
  num_zones: 4
  num_rings_of_interest: 4
  rnr_ver_angle_thr: -15.0
  rnr_intensity_thr: 0.2
  sensor_height: 0.0
  th_seeds: 0.125
  th_dist: 0.125
  th_seeds_v: 0.25
  th_dist_v: 0.1
  max_range: 80.0
  min_range: 0.3
  uprightness_thr: 0.707
  adaptive_seed_selection_margin: -1.2
  num_sectors_each_zone: [16, 32, 54, 32]
  num_rings_each_zone: [2, 4, 4, 4]
  max_flatness_storage: 1000
  max_elevation_storage: 1000
  elevation_thr: [0.0, 0.0, 0.0, 0.0]
  flatness_thr: [0.0, 0.0, 0.0, 0.0]

grid_map_builder:
  grid_res: 0.2

variance_filter:
  variance_th: 0.01
  min_num: 1

first_bgk:
  kernel_radius: 1.0
  lambda: 0.0
  sigma: 1.0
  min_num: 3

trajectory_filter:
  normal_th_deg: 15.0

second_bgk:
  kernel_radius: 2.0
  lambda: 0.0
  sigma: 0.5
  min_num: 3

median:
  enable: true
  kernel_size: 3

divide_by_terrain:
  threshold: 0.1
  on_terrain_only: false

moving_point_identification:
  voxel_leaf_size: 0.2
  fov_h_deg: 360.0
  fov_v_deg: 90.0
  res_h_deg: 0.4
  res_v_deg: 0.9375
  lidar_range: 50.0
  delta_h: 1
  delta_v: 1
  threshold: 0.1
  frame_skip: 0
  submap_update_dist: 5.0
